gadgets/snapshot_socket/test/snapshot_socket_test.go

```diff
@@ -0,0 +1,118 @@
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+type snapshotSocketEntry struct {
+	eventtypes.CommonData
+
+	NetNsID     uint64 `json:"netns_id"`
+	InodeNumber uint64 `json:"ino"`
+
+	SrcEndpoint utils.L4Endpoint `json:"src"`
+	DstEndpoint utils.L4Endpoint `json:"dst"`
+	Status      uint64           `json:"status"`
+}
+
+func TestSnapshotSocket(t *testing.T) {
+	gadgettesting.RequireEnvironmentVariables(t)
+	utils.InitTest(t)
+
+	containerFactory, err := containers.NewContainerFactory(utils.Runtime)
+	require.NoError(t, err, "new container factory")
+	containerName := "test-snapshot-socket"
+	containerImage := "docker.io/library/busybox:latest"
+
+	var ns string
+	containerOpts := []containers.ContainerOption{containers.WithContainerImage(containerImage)}
+
+	if utils.CurrentTestComponent == utils.KubectlGadgetTestComponent {
+		ns = utils.GenerateTestNamespaceName(t, "test-snapshot-socket")
+		containerOpts = append(containerOpts, containers.WithContainerNamespace(ns))
+	}
+
+	testContainer := containerFactory.NewContainer(
+		containerName,
+		"nc -l 0.0.0.0 -p 9090",
+		containerOpts...,
+	)
+
+	testContainer.Start(t)
+	t.Cleanup(func() {
+		testContainer.Stop(t)
+	})
+
+	var runnerOpts []igrunner.Option
+	var testingOpts []igtesting.Option
+	commonDataOpts := []utils.CommonDataOption{utils.WithContainerImageName(containerImage), utils.WithContainerID(testContainer.ID())}
+
+	// TODO: timeout shouldn't be required
+	switch utils.CurrentTestComponent {
+	case utils.IgLocalTestComponent:
+		runnerOpts = append(runnerOpts, igrunner.WithFlags(fmt.Sprintf("-r=%s", utils.Runtime), "--timeout=1"))
+	case utils.KubectlGadgetTestComponent:
+		runnerOpts = append(runnerOpts, igrunner.WithFlags(fmt.Sprintf("-n=%s", ns), "--timeout=1"))
+		testingOpts = append(testingOpts, igtesting.WithCbBeforeCleanup(utils.PrintLogsFn(ns)))
+		commonDataOpts = append(commonDataOpts, utils.WithK8sNamespace(ns))
+	}
+
+	runnerOpts = append(runnerOpts, igrunner.WithValidateOutput(
+		func(t *testing.T, output string) {
+			expectedEntry := &snapshotSocketEntry{
+				CommonData: utils.BuildCommonData(containerName, commonDataOpts...),
+				SrcEndpoint: utils.L4Endpoint{
+					Addr:    "0.0.0.0",
+					Version: 4,
+					Port:    9090,
+					Proto: 6,
+				},
+				DstEndpoint: utils.L4Endpoint{
+					Addr:    "0.0.0.0",
+					Version: 4,
+					Port:    0,
+				},
+				Status:      10,
+				NetNsID:     utils.NormalizedInt,
+				InodeNumber: utils.NormalizedInt,
+			}
+
+			normalize := func(e *snapshotSocketEntry) {
+				utils.NormalizeCommonData(&e.CommonData)
+				utils.NormalizeInt(&e.NetNsID)
+				utils.NormalizeInt(&e.InodeNumber)
+			}
+
+			match.MatchEntries(t, match.JSONSingleArrayMode, output, normalize, expectedEntry)
+		},
+	))
+
+	snashotSocketCmd := igrunner.New("snapshot_socket", runnerOpts...)
+
+	igtesting.RunTestSteps([]igtesting.TestStep{snashotSocketCmd}, t, testingOpts...)
```

Comment:
Just a quick comment because there was a fix for the other `snapshot` gadget: Could you add a sleep before running the snapshot command like it was done in https://github.com/inspektor-gadget/inspektor-gadget/pull/3338